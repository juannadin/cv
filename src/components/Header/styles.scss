.header {
  color: $black;
  background-color: $white;

  &__title {
    margin: calculate-rem(104px) 0 0 0;
  }

  &__subtitle {
    margin: calculate-rem(12px) 0 0 0;
    letter-spacing: $letter-spacing-h2
  }

  &__about, &__motto {
    @include font-size($font-size-22);
    letter-spacing: calculate-rem(0.1px);
    line-height: calculate-rem(33px);
  }

  &__about {
    margin: calculate-rem(53px) 0 0 0;
  }

  &__motto {
    margin: calculate-rem(22px) 0 0 0;
    font-weight: $font-weight-600
  }

  &__info {
    margin-bottom: calculate-rem(38px);
  }

  &__personal-info {
    margin: calculate-rem(47px) 0 0 0;
    display: flex;
    flex-direction: column;

    .personal-info-item {
      @include font-size($font-size-20);
      &__link {
        margin: calculate-rem(4px) 0 calculate-rem(12px) 0;
        display: block;
      }
    }
  }
}

@media (min-width: $desktop-width-break) {
  .header {
    &__personal-info {
      flex-direction: row;
      justify-content: space-between;

      .personal-info-item {
        &__link {
          margin: 0;
          display: inline;
          margin-left: calculate-rem(8px);
        }
      }
    }
  }
}
