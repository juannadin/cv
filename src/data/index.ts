
export type Data = {
  header: Header;
  body: Body;
};

export type Header = {
  title: string;
  subtitle: string;
  about: string;
  motto: string;
  personalInfo: Array<PersonalInfo>;
}

export type Body = {
  skills: Skills;
  experiences: Experiences;
  educations: Educations;
  extraMiles: ExtraMiles;
  interests: Interests;
}

export type Skills = {
  soft: Array<Skill>;
  technical: Array<Skill>;
  languajes: Array<Skill>;
}

export type Experiences = Array<Experience>

export type Educations = Array<Education>

export type ExtraMiles = Array<string>

export type Interests = Array<string>

type PersonalInfo = {
  type: string;
  value: string;
};

type Skill = {
  name: string;
  ammount: number;
};

type Experience = {
  company: string;
  title: string;
  description: string;
  startDate: string;
  endDate: string;
};

type Education = {
  institution: string;
  title: string;
  degree: string;
  startDate: string;
  endDate: string;
};

const data: Data = {
  header: {
    title: 'Juan Ignacio Nadin.',
    subtitle: '[ Product Owner & Full Stack Developer ]',
    about: 'I\'m a data-driven, product-oriented and results-focused person, with +10 years of experience in the software industry. I understand that technology is no limitation for product design.',
    motto: 'There\'s always a way.',
    personalInfo: [
      {
        type: 'Address',
        value: 'Martinez, Buenos Aires, ARG',
      },
      {
        type: 'Email',
        value: 'juan.ignacio.nadin@gmail.com',
      },
      {
        type: 'Cell',
        value: '+54 11 6332-2736',
      },
    ],
  },
  body: {
    skills: {
      soft: [
        {
          name: 'Leadership',
          ammount: 90,
        },
        {
          name: 'Team Work',
          ammount: 85,
        },
        {
          name: 'Decision-making',
          ammount: 90,
        },
        {
          name: 'Time Management',
          ammount: 80,
        },
        {
          name: 'Communication',
          ammount: 95,
        },
        {
          name: 'Agile Development',
          ammount: 80,
        },
      ],
      technical: [
        {
          name: 'Node.js',
          ammount: 95,
        },
        {
          name: 'Express.js',
          ammount: 90,
        },
        {
          name: 'React.js',
          ammount: 90,
        },
        {
          name: 'Typescript',
          ammount: 80,
        },
        {
          name: 'AWS',
          ammount: 75,
        },
        {
          name: 'Unit & e2e testing',
          ammount: 95,
        },
      ],
      languajes: [
        {
          name: 'Spanish',
          ammount: 100,
        },
        {
          name: 'English',
          ammount: 90,
        },
        {
          name: 'Portuguesse',
          ammount: 20,
        },
      ],
    },
    experiences: [
      {
        company: 'Mercado Libre',
        title: 'Senior Full Stack Developer',
        description: 'Full ownership of the entire development process. Product Desgin, strategic thinking, estimations and allocations, scrum agile development, production build and release, business tracking. Mentoring, asset hunting and screening.',
        startDate: 'Nov 2016',
        endDate: 'Present',
      },
      {
        company: 'Wunderman',
        title: 'Technology Lead',
        description: 'Responsible of leading a development team. Leadership and management skills applied in a technical context with a developers squad. Projects estimations, mentoring and guidance, work stream and allocation',
        startDate: 'Sep 2015',
        endDate: 'Nov 2016',
      },
      {
        company: 'Wunderman',
        title: 'Senior Front End Developer',
        description: 'Front-End web development for several clients such as ford.com, and lincoln.com. Javascript, HTML, CSS, XML, Freemarker and AEM plattforms, Through SVN & GIT. Desktop & Mobile.',
        startDate: 'Jun 2012',
        endDate: 'Aug 2015',
      },
      {
        company: 'IBM',
        title: 'Portal Web Developer',
        description: 'BT/IT PartnerWorld Web Portal Producer. Maintenance and frontend development front-end IBM internal sites through CMS',
        startDate: 'Dec 2010',
        endDate: 'May 2012',
      },
    ],
    educations: [
      {
        institution: 'Escuela de Arte Multimedial Da Vinci',
        title: 'Web Design & Programming',
        degree: 'Technical degree',
        startDate: '2009',
        endDate: '2012',
      },
      {
        institution: 'Escuela San Ladislao',
        title: 'Economy and Management',
        degree: 'High School degree',
        startDate: '2002',
        endDate: '2004',
      },
    ],
    extraMiles: [
      'User Experience',
      'Jira, Trello',
      'Adobe Illustrator & Photoshop',
      'Zeplin',
      'SQL, AEM, PHP, jQuery',
    ],
    interests: [
      'Music production',
      'Composition',
      'Guitar and Piano',
      'Parenthood',
      'Volleyball',
    ],
  }
};

export default data;
