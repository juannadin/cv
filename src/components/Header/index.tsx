import React, { FC } from 'react';
import { Header as HeaderProps } from '../../data';
import { PersonalInfo, PersonalInfoType } from '../../data';

const getLink = (element: PersonalInfo) => {
  const cases = {
    [PersonalInfoType.ADDRESS]: 'https://goo.gl/maps/1iYrzpWL5EnDp2GT7',
    [PersonalInfoType.EMAIL]: `mailto:${element.value}`,
    [PersonalInfoType.CELLPHONE]: `tel:${element.value}`,
  };

  return cases[element.type];
}

const Header: FC<HeaderProps> = ({ title, subtitle, about, motto, personalInfo }) => (
  <header className="header">
    <div className="container">
      <h1 className="header__title">{title}</h1>
      <h2 className="header__subtitle">{subtitle}</h2>
      <p className="header__about">{about}</p>
      <p className="header__motto">{motto}</p>
      <address className="header__info">
        <ul className="header__personal-info">
          {personalInfo.map((element, key) => {
            const link = getLink(element);
            return(
              <li className="personal-info-item" key={key}>
                <strong className="personal-info-item__type">{element.type}</strong>
                <a className="personal-info-item__link" href={link} target="_blank" rel="noreferrer">
                  {element.value}
                </a>
              </li>)
          })}
        </ul>
      </address>
    </div>
  </header>
);

export default Header;
