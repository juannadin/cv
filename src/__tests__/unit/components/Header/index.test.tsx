import React from 'react';
import { render } from '@testing-library/react';
import Header from '../../../../components/Header/index';
import { Header as HeaderProps, PersonalInfoType } from '../../../../data';

const headerProps: HeaderProps = {
  title: 'this is a title',
  subtitle: 'this is a subtitle',
  about: 'lorem ipsum about content',
  motto: 'this is my motto',
  personalInfo: [
    {
      type: PersonalInfoType.ADDRESS,
      value: 'Martinez, Buenos Aires, ARG',
    },
    {
      type: PersonalInfoType.EMAIL,
      value: 'juan.ignacio.nadin@gmail.com',
    },
    {
      type: PersonalInfoType.CELLPHONE,
      value: '+54 11 6332-2736',
    },
  ],
};

describe('Header', () => {
  test('render all components', () => {
    const { container } = render(<Header {...headerProps}/>);
    expect(container).toMatchSnapshot();
  });
})

