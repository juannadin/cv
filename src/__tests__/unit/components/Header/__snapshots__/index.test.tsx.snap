// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header render all components 1`] = `
<div>
  <header
    class="header"
  >
    <div
      class="container"
    >
      <h1
        class="header__title"
      >
        this is a title
      </h1>
      <h2
        class="header__subtitle"
      >
        this is a subtitle
      </h2>
      <p
        class="header__about"
      >
        lorem ipsum about content
      </p>
      <p
        class="header__motto"
      >
        this is my motto
      </p>
      <address
        class="header__info"
      >
        <ul
          class="header__personal-info"
        >
          <li
            class="personal-info-item"
          >
            <strong
              class="personal-info-item__type"
            >
              Address
            </strong>
            <a
              class="personal-info-item__link"
              href="https://goo.gl/maps/1iYrzpWL5EnDp2GT7"
              rel="noreferrer"
              target="_blank"
            >
              Martinez, Buenos Aires, ARG
            </a>
          </li>
          <li
            class="personal-info-item"
          >
            <strong
              class="personal-info-item__type"
            >
              Email
            </strong>
            <a
              class="personal-info-item__link"
              href="mailto:juan.ignacio.nadin@gmail.com"
              rel="noreferrer"
              target="_blank"
            >
              juan.ignacio.nadin@gmail.com
            </a>
          </li>
          <li
            class="personal-info-item"
          >
            <strong
              class="personal-info-item__type"
            >
              Cell
            </strong>
            <a
              class="personal-info-item__link"
              href="tel:+54 11 6332-2736"
              rel="noreferrer"
              target="_blank"
            >
              +54 11 6332-2736
            </a>
          </li>
        </ul>
      </address>
    </div>
  </header>
</div>
`;
